systemd/cephfs-umount.service.in

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Unmount CephFS before shutting down the system
+After=network-online.target ceph.target
+Before=remote-fs-pre.target
+Wants=network-online.target remote-fs-pre.target ceph.target
+
+[Service]
+EnvironmentFile=-@SYSTEMD_ENV_FILE@
+Type=oneshot
+RemainAfterExit=yes
+ExecStop=/usr/bin/umount -lf -a -t ceph
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
This doesn't seem to follow regular systemd mount unit configuration.
Why can't we rely on regular systemd fstab handling (with `_netdev` to flag the network dependency)? Is it just the `ceph.target` ordering dependency? If so, could this be achieved with `x-systemd.before/after`?

Reply:
> This doesn't seem to follow regular systemd mount unit configuration.

Yes, totally agree, but case doesn't always go ideally...

In my case due to the ordering conflict between ceph/etcd/crio/kublet and `remote-fs-pre.target`/`remote-fs.target`, my libceph always complain about missing ceph mon during unmount. 

Therefore I need a entrypoint to secure all CephFS unmount cleanly **BEFORE** `ceph.target` goes down.

With this PR now we have:
- `_netdev` unmount by systemd fstab during `remote-fs.target`
- Between `remote-fs-pre.target` and `ceph.target`, `cephfs-umount.service` act as a fail-safe to lazy unmount any possible remaining CephFS mount point
- Now `ceph.target` could able to shutdown cleanly

> If so, could this be achieved with x-systemd.before/after?

We better not do so, because even `systemctl restart ceph.target` will also trigger the fail-safe lazy unmount of running CephFS; with `cephfs-umount.service` it only activate during system graceful reboot/shutdown.

Reply:
@ddiss I will give a try again with my production cluster, to see if we could skip `cephfs-umount.service` and still without reboot deadlock.

Reply:
@ddiss Ok if ALL `ceph.target` and its child services could stop AFTER `remote-fs.target` and `remote-fs-pre.target`, no graceful shutdown/reboot deadlock happened. I will remove the `cephfs-umount.service` from PR accordingly:

![Screenshot from 2020-09-17 12-57-57](https://user-images.githubusercontent.com/780562/93422042-7f96a600-f8e5-11ea-9db8-a0af1cfd8929.png)